<launch>
  <!-- subscribes to cmd_vel topic -->
  <!-- publishes odom topic with base_link frame -->
  <!-- broadcasts odom->base_link transform -->
  <arg name="param" default="value" />
  <node pkg="roboteq_diff_driver" type="driver" name="roboteq_diff_driver" output="screen">
    <!-- enable broadcast of odom tf -->
    <param name="pub_odom_tf" value="true" />
    <!-- specify odom frame -->
    <param name="odom_frame" value="odom" type="str"/>
    <!-- specify base frame -->
    <param name="base_frame" value="base_link" type="str"/>
    <!-- specify cmd_vel topic -->
    <param name="cmdvel_topic" value="/tender_bot/cmd_vel" type="str"/>
    <!-- specify port for roboteq main wheels controller -->
    <param name="port" value="/dev/TenderWheels" type="str"/>
    <!-- specify baud for roboteq main wheels controller -->
    <param name="baud" value="115200" />
    <!-- specify whether to use open-loop motor speed control (as opposed to closed-loop)-->
    <param name="open-loop" value="false" />
    <!-- specify robot wheel circumference in meters  -->
    <param name="wheel_circumference" value="0.26" />
    <!-- specify robot track width in meters  -->
    <param name="track_wdith" value="1.1" />
    <!-- specify pulse-per-revolution for motor encoders  -->
    <param name="encoder_ppr" value="1000" />
    <!-- specify counts-per-revolution for motor encoders (ppr*4 for quadrature encoders)  -->
    <param name="encoder_cpr" value="4000" />
    <!-- specify port for roboteq jockey wheel controller -->
    <param name="port2" value="/dev/TenderJockey" type="str"/>
    <!-- specify baud for roboteq jockey wheel controller -->
    <param name="baud2" value="115200" />
    <!-- specify whether to use open-loop motor speed control (as opposed to closed-loop)-->
    <param name="open-loop2" value="true" />
    <!-- specify pulse-per-revolution for motor encoders  -->
    <param name="encoder_ppr2" value="1000" />
    <!-- specify counts-per-revolution for motor encoders (ppr*4 for quadrature encoders)  -->
    <param name="encoder_cpr2" value="4000" />
  </node>
</launch>
